app/services/fastly_config/base.rb

```diff
@@ -0,0 +1,35 @@
+module FastlyConfig
+  class Base
+    attr_reader :fastly, :active_version, :updated_files
+
+    def initialize(fastly, active_version)
+      @fastly = fastly
+      @active_version = active_version
+      @updated_files = get_updated_files
+    end
+
+    def update_needed?
+      updated_files.present?
+    end
+
+    def update(new_version)
+      updated_files.each { |filename| update_config(new_version, filename) }
+    end
+
+    private
+
+    def get_updated_files(files: self.class::FASTLY_FILES)
+      updated_files = []
+      Dir.glob(files).each { |filename| updated_files << filename if file_updated?(filename) }
```

Comment:
Ruby recently added `filter_map` which performs a mapping and removes elements that would be `nil` afterward so you don't have to create an array first and push into it.

```ruby
Dir.glob(files).filter_map { |filename| filename if file_updated?(filename) }
```

Reply:
That's awesome! TIL 🎉